tests/CMakeLists.txt

```diff
@@ -169,18 +169,28 @@ FOREACH(_test ${_tests})
       SET(_testlib)
     ENDIF()
 
-    # Create the output directory and a rule to generate the input file therein.
-    # This input file is the one found in the source dir, but it has the
-    # correct output directory so that we do not need to specify it by hand
-    # (a common mistake when copying a previous test) and do not need to
-    # specify any additional plugin shared libraries, should they exist
-    # for this test
+    # Create the output directory and subdirectories if necessary
+    FILE(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/output-${_test})
+
+    FILE(GLOB_RECURSE _outputs RELATIVE ${CMAKE_CURRENT_SOURCE_DIR}/${_test}/
+	${CMAKE_CURRENT_SOURCE_DIR}/${_test}/[a-zA-Z0-9]*)
+    FOREACH(_output ${_outputs})
+      GET_FILENAME_COMPONENT(_dir ${_output} DIRECTORY)
+      IF(NOT  ${_dir} STREQUAL "" )
```

Comment:
Oh, that's the part that I was missing. Thanks for pointing this out. Maybe add a comment of the sort
```
# Recursively get all files (and only files, not subdirectories) located in this test's directory. 
# For each file get the directory part and create a corresponding subdirectory in the
# output directory.
```